heartbeat/heartbeat.py

```diff
@@ -1,40 +1,49 @@
 import sys
-
-if sys.version > '3':
-  from http.server import HTTPServer
-  from http.server import BaseHTTPRequestHandler
-else:
-  from BaseHTTPServer import HTTPServer
-  from BaseHTTPServer import BaseHTTPRequestHandler
-
 import json
 import threading
 import datetime
+import subprocess
 
-import git_sha
+if sys.version > '3':
+    from http.server import HTTPServer
+    from http.server import BaseHTTPRequestHandler
+else:
+    from BaseHTTPServer import HTTPServer
+    from BaseHTTPServer import BaseHTTPRequestHandler
 
 global starting_time
 starting_time = None
 
+
 class HeartbeatRequestHandler(BaseHTTPRequestHandler):
-  def do_GET(self) :
-    global starting_time
-    if self.path == '/heartbeat' :
-      self.send_response(200)
-      self.send_header('Content-type:', 'text/html')
-      now = datetime.datetime.now()
-      uptime = str(now - starting_time)
-      obj = {'status': 'running', 'build': git_sha.git_sha, 'uptime': uptime}
-      self.wfile.write(('\n' + json.dumps(obj) + "\n").encode('utf-8'))
+    def do_GET(self):
+        global starting_time
+        if self.path == '/heartbeat':
+            self.send_response(200)
+            self.send_header('Content-type:', 'text/html')
+            now = datetime.datetime.now()
+            uptime = str(now - starting_time)
+            try:
+                import git_sha
+                commit_hash = git_sha.git_sha
+            except ImportError:
+                try:
+                    commit_hash = subprocess.check_output(["git", "rev-parse", "HEAD"])
+                except subprocess.CalledProcessError:
+                    commit_hash = 'NA'
```

Comment:
I think in this we should fail here. 

If there is no git installed (probably development environment) and no `git_sha.py` file (which should exist in prod environment) something is inconsistent. 

It would allow to run in production (due to misconfiguration, for instance) without git SHA1 hash.